sql/src/main/java/io/crate/operation/collect/MapSideDataCollectOperation.java

```diff
@@ -288,17 +288,21 @@ private CollectService getCollectService(CollectPhase collectNode, String localN
                             projectorChain.startProjections(jobCollectContext);
                             localRowDownStream = projectorChain.firstProjector();
                         }
-                        CrateCollector collector = collectService.getCollector(localCollectNode, localRowDownStream); // calls projector.registerUpstream()
-                        collector.doCollect();
                     }
+                    CrateCollector collector = collectService.getCollector(localCollectNode, localRowDownStream); // calls projector.registerUpstream()
+                    crateCollectors.add(collector);
```

Comment:
could this cause issues due to threading? if kill is called before the callable is started the list might still be empty